### PR TITLE
Add NXP i.MX8 SOC family support.

### PR DESCRIPTION
**Pull request by unbel13ver**
Currently there are two devices supported:
* imx8qm-mek
* imx8qxp-mek

Signed-off-by: Ivan Nikolaenko <ivan.nikolaenko@unikie.com>

###### Description of changes
Changes made in the way it makes very easy to add another mx8-based device or device-specific configuration.

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

**Comment by necrophcodr** [comment]
@unbel13ver is the firmware files completely public domain to use, or are they under some license agreement that the user needs to agree to?

**Comment by Mic92** [comment]
I think in either case we would be fine to have the nix expression in this repository because the binaries are public accessible and we are not hosting repackaged versions of them (hence no copy). Might be nice though to have a license in `meta.license` if available.

**Comment by unbel13ver** [comment]
> @unbel13ver is the firmware files completely public domain to use, or are they under some license agreement that the user needs to agree to?

Thanks for quick reply. Yes, the firmware is publicly available, but one needs to accept NXP's EULA:
> Welcome to NXP imx-seco-3.8.6.bin
> 
> You need to read and accept the EULA before you can continue.
> 
> LA_OPT_NXP_Software_License v34 February 2022

I am not quilte familiar with legal stuff, so I would appreciate a lot if you point me what kind of license information I should add.

**Comment by Mic92** [comment]
Does not look like it's redistributable so I would go with `lib.licenses.unfree`

**Comment by unbel13ver** [comment]
> Does not look like it's redistributable so I would go with `lib.licenses.unfree`

Addressed

**Comment by Mic92** [comment]
bors merge

File: nxp/imx8qm-mek/overlay.nix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+final: _prev: {
+  inherit ( final.callPackage ./bsp/u-boot/imx8/imx-uboot.nix { pkgs = final; targetBoard = "imx8qm"; }) ubootImx8 imx-firmware;
```
Comment (tattvam) [inline nxp/imx8qm-mek/overlay.nix:2]: Is `./bsp/u-boot/imx8/imx-uboot.nix ` correct path?

File: nxp/imx8qxp-mek/overlay.nix [hunk 2]
```diff
@@ -0,0 +1,3 @@
+final: prev: {
+  inherit ( final.callPackage ./bsp/u-boot/imx8/imx-uboot.nix { pkgs = final; targetBoard = "imx8qxp"; }) ubootImx8 imx-firmware;
```
Comment (tattvam) [inline nxp/imx8qxp-mek/overlay.nix:2]: `./bsp/u-boot/imx8/imx-uboot.nix` no u-boot/imx8 directory in basp. Also relative path of bsp doesn't seem ok.

File: nxp/imx8qm-mek/overlay.nix (hunk 1 above)
Comment (unbel13ver) [reply nxp/imx8qm-mek/overlay.nix:2]: Hi, sorry for this. I'll fix paths and improve README

File: nxp/imx8qxp-mek/overlay.nix (hunk 2 above)
Comment (unbel13ver) [reply nxp/imx8qxp-mek/overlay.nix:2]: Relative paths are ok as long as they point within the repo directory structure.

**Pull request merged**